crates/core/src/db/datastore/locking_tx_datastore/datastore.rs

```diff
@@ -543,7 +543,15 @@ impl MutTxDatastore for Locking {
     }
 }
 
-pub(super) fn record_metrics(ctx: &ExecutionContext, tx_timer: Instant, lock_wait_time: Duration, committed: bool) {
+/// This utility is responsible for recording all transaction metrics.
+pub(super) fn record_metrics(
+    ctx: &ExecutionContext,
+    tx_timer: Instant,
+    lock_wait_time: Duration,
+    committed: bool,
+    tx_data: Option<&TxData>,
+    committed_state: Option<&CommittedState>,
```

Comment:
`TxData` is needed for recording inserts and deletes. `CommittedState` is needed for recording table rows and table size.